crates/ruff/src/checkers/ast/analyze/for_loops.rs

```diff
@@ -0,0 +1,33 @@
+use rustpython_parser::ast::{self, Stmt};
+
+use crate::checkers::ast::Checker;
+use crate::codes::Rule;
+use crate::rules::{flake8_bugbear, perflint};
+
+/// Run lint rules over all deferred for-loops in the [`SemanticModel`].
+pub(crate) fn for_loops(checker: &mut Checker) {
```

Comment:
Nit: 
```suggestion
pub(crate) fn deferred_for_loops(checker: &mut Checker) {
```

I was confused why the function has no node argument.